packages/zapp/console/src/components/Executions/Tables/NodeExecutionsTable.tsx

```diff
@@ -41,13 +43,26 @@ export const NodeExecutionsTable: React.FC<NodeExecutionsTableProps> = ({
   const commonStyles = useCommonStyles();
   const tableStyles = useExecutionTableStyles();
   const nodeExecutionsById = useContext(NodeExecutionsByIdContext);
+  const filterState = useNodeExecutionFiltersState();
+
+  const showUnknownNodes = useMemo<Boolean>(() => {
+    if (
+      filterState.filters[0].active &&
+      !(filterState.filters[0] as MultiFilterState<any, any>).selectedStates[
+        NodeExecutionPhase.UNDEFINED
+      ]
+    ) {
+      return false;
+    }
+    return !filterState.filters[1].active && !filterState.filters[2].active;
+  }, [filterState]);
```

Comment:
this is not a scalable solution - it does the fix, but going to be a disaster to maintain, if/when we add more filters.

In `ExecutionNodeViews` we pass `appliedFilters` as a query param, maybe we can implement something similar here